packages/core/user-settings/core-user-settings-server-internal/src/user_settings_service.ts

```diff
@@ -0,0 +1,73 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import type { CoreContext } from '@kbn/core-base-server-internal';
+import type { Logger } from '@kbn/logging';
+import type { KibanaRequest } from '@kbn/core-http-server';
+import type { DarkModeValue } from '@kbn/core-ui-settings-common';
+import type { InternalUserProfileServiceStart } from '@kbn/core-user-profile-server-internal';
+
+export interface UserSettingsServiceStartDeps {
+  userProfile: InternalUserProfileServiceStart;
+}
+
+const userSettingsDataPath = 'userSettings';
+
+/**
+ * @internal
+ */
+export interface InternalUserSettingsServiceSetup {
+  getUserSettingDarkMode: (request: KibanaRequest) => Promise<DarkModeValue | undefined>;
+}
+
+/**
+ * @internal
+ */
+export class UserSettingsService {
+  private logger: Logger;
+  private userProfile?: InternalUserProfileServiceStart;
```

Comment:
FWIW, this is not a new file, it was moved from `packages/core/user-settings/core-user-settings-server-internal/user_settings_service.ts`, but I guess the file changed too much so it lost the info